src/tensorlake/function_executor/proto/function_executor.proto

```diff
@@ -138,6 +159,54 @@ message RunTaskResponse {
     optional SerializedObject invocation_error_output = 10;
 }
 
+// TaskRequest contains the input data for task execution
+message TaskRequest {
+    optional SerializedObject function_input = 1;
+    optional SerializedObject function_init_value = 2;
+}
+
+// TaskResult contains the execution outcome and outputs of a task
+message TaskResult {
+    repeated SerializedObject function_outputs = 1;
+    // Edges that the function wants the invocation to be routed to.
+    repeated string next_functions = 2;
+    optional string stdout = 3;
+    optional string stderr = 4;
+    optional bool is_reducer = 5;
+    optional Metrics metrics = 6;
+    optional TaskOutcomeCode outcome_code = 7;
+    optional TaskFailureReason failure_reason = 8;
+    // User payload for invocation error if task failed with invocation error.
+    optional SerializedObject invocation_error_output = 9;
+}
+
+// Task represents a task with its metadata and execution result
+message Task {
+    optional string task_id = 1;
+    optional string namespace = 2;
+    optional string graph_name = 3;
+    optional string graph_version = 4;
+    optional string function_name = 5;
```

Comment:
Why is the function name going to the FE? If these are used for reconciling state of the FE if the executor loses connection to the FE or if it's restarted, isn't task id enough?

Reply:
The function name is here because it's used in RunTaskRequest, and the Task message is also used in the CreateTaskRequest - this message type isn't solely for reconnecting after restart.